docker/Dockerfile

```diff
@@ -32,6 +33,15 @@ RUN apt-get update -qq \
     && apt-get install -y nodejs \
     && rm -rf /var/cache/apt/* /var/lib/apt/lists/*;
 
+# Add the necessary apt-gets for devcontainer if this is a devcontainer
+RUN if [ -n "$DEVCONTAINER" ] ; then \
+    apt-get update -qq \
+        && apt-get install -y \
+            clang-format-14 \
+            clang-tidy; fi
+
```

Comment:
Why's this needed in the devcontainer, but not in the docker image?

Reply:
The docker image itself is used effectively as a compiler. The container is built, then the compile commands are run and the container is destroyed again. When developing inside a devcontainer the container stays up as the development enviromment.
Since the container doubles as a dev environment we need all tools normally used on the host for "normal" development on the host, like clang-format